pilot/pkg/proxy/envoy/v2/pushqueue.go

```diff
@@ -0,0 +1,83 @@
+package v2
+
+import (
+	"sync"
+
+	"istio.io/istio/pilot/pkg/model"
+)
+
+type PushInformation struct {
+	// If not empty, it is used to indicate the event is caused by a change in the clusters.
+	// Only EDS for the listed clusters will be sent.
+	edsUpdatedServices map[string]struct{}
+
+	push *model.PushContext
+
+	full bool
+}
+
+type PushQueue struct {
+	mu          sync.RWMutex
+	connections map[*XdsConnection]*PushInformation
+	order       []*XdsConnection
+	signal      chan struct{}
+}
+
+func NewPushQueue() *PushQueue {
+	return &PushQueue{
+		connections: make(map[*XdsConnection]*PushInformation),
+		signal:      make(chan struct{}),
+	}
+}
+
+func (p *PushQueue) Add(proxy *XdsConnection, pushInfo *PushInformation) {
```

Comment:
This is the important part of the change.
Please add explanation, like
1. Order in which a client is served is decided when a proxy enters push queue for the 1st time.
2. If proxy enters queu again, only push details are updated but the client retains its spot.

Reply:
I updated the docs now. Thought I had them already - I think I accidentally pushed over a commit I had that added comments :slightly_frowning_face: